Added mechanism to push device package to list of devices

ppal-zededa opened the pull request:
Added mechanism to push device package to list of devices
Added script to install XEN hypervisor and bring up Dom0 on top of virtualbox VM

ppal-zededa committed:
Added mechanism to push device package to list of devices
Added script to install XEN hypervisor and bring up Dom0 on top of virtualbox VM

ckzed reviewed (commented):
The device name enp0s3/enp0s8 is not consistent across all devices. The naming depends on the PCI Express slot that is used. If the idea was to extract the ethernet device name on which the default gateway is configured, then one could use "ip route show | grep default" to get the device name.

The other thing was the issue that we discussed already: the files are being copied to home directory which might not exist on the device.

ppal-zededa committed:
Used /tmp directory instead of home directory for copying debian packages
Added support for making package available for zupgrade

ppal-zededa committed:
Handled Pull model as well.
Once zeded-provision debian package is built,
it will be available for wget in /opt/zededa/debian directory

saurabh-zededa committed:
prompt for reboot and dynamic listing of interfaces.

ppal-zededa committed:
Modified reading INSTALL_DEVICE_LIST as include parameter
By default INSTALL_DEVICE_LIST will be empty

ppal-zededa committed:
Merge branch 'ckzed-url' of https://github.com/zededa/go-provision into ckzed-url
